crates/compiler/parse/src/type_annotation.rs

```diff
@@ -605,7 +605,10 @@ fn expression<'a>(
         .parse(arena, state, min_indent)?;
 
         let (p2, rest, rest_state) = zero_or_more(skip_first(
-            backtrackable(byte(b',', EType::TFunctionArgument)),
+            backtrackable(skip_first(
```

Comment:
Rather than doing `skip_first`, I think we need to be preserving this space in the tree. Otherwise we could accidentally lose a comment when formatting, for example.

Rather than separately tracking spaces before/after the comma, we should probably just merge them. There's a handy `merge_spaces` function for just that purpose.